packages/commercetools/composables/src/useCart/index.ts

```diff
@@ -50,11 +50,11 @@ const useCartFactoryParams: UseCartFactoryParams<CartDetails, LineItem, ProductV
   clear: async (context: Context, { currentCart }) => {
     return currentCart;
   },
-  applyCoupon: async (context: Context, { currentCart, couponCode, customQuery }) => {
+  applyCoupon: async (context: Context, { currentCart, coupon, customQuery }) => {
     const cartDetails = await getCurrentCartDetails(context, currentCart);
 
-    const { data } = await context.$ct.api.applyCartCoupon(cartDetails, couponCode, customQuery);
-    return { updatedCart: data.cart, updatedCoupon: couponCode };
+    const { data } = await context.$ct.api.applyCartCoupon(cartDetails, coupon, customQuery);
+    return { updatedCart: data.cart, updatedCoupon: coupon };
   },
   removeCoupon: async (context: Context, { currentCart, coupon, customQuery }) => {
```

Comment:
The argument in both the `applyCoupon` and the `removeCoupon` method should be named `couponCode`, so it's obvious that we expect a code, not a Coupon object.

**The `applyCoupon` method doesn't change.** The `removeCoupon`, however, should accept `couponCode` instead of `coupon`. You can use the `currenctCart` object to extract the Coupon object based on the code and then get the `coupon.id` used in the method.

Reply:
fixed this problem, and changed the migrate to 2.4